plugin/src/main/java/net/dzikoysk/funnyguilds/user/DamageCache.java

```diff
@@ -1,42 +1,171 @@
 package net.dzikoysk.funnyguilds.user;
 
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import java.time.Duration;
 import java.time.Instant;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import net.dzikoysk.funnyguilds.FunnyGuilds;
+import net.dzikoysk.funnyguilds.config.PluginConfiguration.DamageTracking;
+import net.dzikoysk.funnyguilds.shared.MapUtils;
+import panda.std.Option;
 
 public class DamageCache {
 
-    private final User attacker;
+    private final User user;
 
-    private double damage;
-    private Instant lastTime;
+    private final LinkedList<Damage> damageHistory = new LinkedList<>();
+    private final Cache<UUID, Instant> killHistory = CacheBuilder.newBuilder()
```

Comment:
Przydałoby sie wymienić ten cache na zwykłą mape. Jeśli pobierzemy z niego wygasniętą wartość to zostanje ona zwrócona ale przy okazji usunieta z cache. Więc dla każdego 2n - 1 (I want to be smort) razu ranking zostanie naliczony.